### ISSUE TITLE
Cross platform LV2 builds

### ISSUE THREAD

**Issue by dromer** [surge-synthesizer/surge#3768]
There is no reason to have LV2 for Linux only since it's a cross-platform plugin format.

Since Reaper will get get LV2 support in the next release, Zrythm almost ready for 1.0 and Ardour, Carla and Element already out there in the wild, there is now a growing amount of LV2 hosts for windows and mac.

Therefore it would be nice if Surge was also available in this format for these other platforms.

**Comment by baconpaul** [comment]
Hi!

The LV2 is buggy and doesn’t work for a few technical reasons described both on our home page and in the issues tagged “LV2” here.  We won’t be investing in this format in the current code base now VST3 is a credible cross platform format (although we are contemplating a port to juce, which I understand may make LV2 viable again).

We recommend everyone use the VST3 on all platforms.

**Comment by dromer** [comment]
Thnx for the quick reply.

I've been using the LV2 on Linux today, fresh out of the master branch.
Works fine, new skin looks awesome. Haven't found any bugs.

Will search for those issues and see if they are still relevant.

**Comment by baconpaul** [comment]
Change an FX type with automation latch on in ardour. It will fail frequently.

The problem is that the surge DSP engine can change parameters (since it is in effect a multi-effect so daw automating one parameter changes the value name and meaning of another). The traditional LV2 forbids this since ports are one-way only. The use of ‘atoms’ which are bidirectional stumped the LV2 team who worked on it (which included falktx).

Strongly recommend you use the VST3. It is both more tested and more robust in several known situations.

**Comment by baconpaul** [comment]
https://github.com/surge-synthesizer/surge/issues/1577 that issue has an awful lot of the convo.

**Comment by dromer** [comment]
Ok, thnx for the info.

A juce port would mean you are on your own for LV2 (or use one of the forks, like falktx' DISTRHO-JUCE) and would likely still not solve your issue with the ports. Of course this could still be done with a new atom type perhaps, but I don't know how deep the rabbit hole you'd have to go before you see drobilla waving a big "STOP"-sign.

**Comment by baconpaul** [comment]
Yeah

Look our strategy has been to work with the DAW manufacturers on VST3 on LInux. As of 1.8 we have confirmed that Surge VST3 Linux works in almost every environment we’ve found and have worked with the devs at Ardour, QTractor, Bitwig and elswhere as we found bugs in their DAWs and we exposed bugs in Surge (and in VST3 (and in VSTGUI (especially in VSTGUI))). If you use the most recent version of almost any linux DAW you can run the VST3 (I think there’s one exception maybe?), and moreover Juce 6 supporting LInux VST3 has made our other plugs work seamlessly (which is one of many reasons we want to port to juce).

My guess is sometime in April we rm -rf src/lv2 (as well as src/vst3, src/vst2 and src/au as well as the vstgui.surge and vst3sdk submodules) in favor of a newly-named plugin with the surge engine and UI built in JUCE, and then go from there.

**Comment by dromer** [comment]
Ok! so that would be a 2.0 effort basically. Good luck with that!

I'm ok with using the vst3 for now. An lv2 would be particularly nice to have for use in embedded systems like the MOD, Zynthian or Poly digit/beepo type of environments. That would be my main use for it personally (and to promote a FOSS plugin format as well of course).

**Comment by baconpaul** [comment]
Yeah the zynthian guys are using the LV2 and they are not as exposed to the problems because they don’t have the same automation framework. I gave em a hand (and also cleaned up the PI port).

as much as it pains me to say something nice about VST3, which has been a thorn in my side for a while, it is GPL3 software also (with a dual license strategy for commercial plugins), so it is a FOSS compatible plugin format.

**Comment by baconpaul** [comment]
Oh and the “2.0” port (which we are calling “Surge XT” because “2.0” never ships) is actually starting to appear in the codebase if you look closely. Our rough plan (which we keep on the pinned ‘status’ issue here) is to ship a 1.8.2 this spring then heads down to finish the juce port, but I’m doing it in parallel in the same code base, not forked. Thanks for the wishes of luck!!!

**Comment by dromer** [comment]
I'd say it's OSS and not FOSS, but that's maybe nitpicking. Either way I prefer a community effort and not Steinberg/Yamaha calling the shots on how software is supposed to work (and what I hear from devs is that vst3-sdk is pretty horrid on the inside).

Also LV2 is extensible so you can implement more advanced features and things like novel control interfaces without going into corporate lobbying and waiting for VST4 to appear.

(just my 2ct :) )

**Comment by dromer** [comment]
(if JUCE doesn't work out you could try DPF .. although there isn't vst3 in there .. yet)

**Comment by baconpaul** [comment]
Yup I hear you. The music industry has made some really odd choices on standards governance. As a relative newcomer to the space, I am surprised with how it all works.

Anyway my rough plan is “move to JUCE and never think about it again”; JUCE is very high quality software, we’ve had great success with it. And I’ve been deep inside vst3sdk and while I don’t say bad things about other software on public forums often, I would not push back on your characterization.

**Comment by baconpaul** [comment]
> (if JUCE doesn't work out you could try DPF .. although there isn't vst3 in there .. yet)

The vast majority of our users are windows and macos. VST3 and AU are a hard requirement for us. But also: JUCE is great and works wonderfully.

**Comment by dromer** [comment]
I'd say "JUCE works" .. wonderfully depends wholly on whoever is building with it :)

I think by now JUCE is used by something like ~90% of audio projects out there .. but only a minority realize that you can compile for other platforms than win/mac.

Anyway, lets stop this slow-chat. Hang out in #lad on irc.freenode.net if you want to chat with knowledgeable FOSS audio devs :)

**Comment by baconpaul** [comment]
Yeah I’m always surprised more juce devs don’t ship linux.
And you are all welcome on our discord any time you want to talk with surge team!

**Comment by dromer** [comment]
Uhg, I'm getting very tired of "yet another discord server" with a dozen more channels and more fragmentation :#

**Comment by baconpaul** [comment]
Well you are also welcome to not join it :) Do whatever works for you!
